File: src/Application/Accountant.Application.Fs/Models/Sync.fs
namespace Accountant.Application.Fs.Models

open System
open Accounts
open AutomaticTransactions
open Categories
open Debts
open Transactions
open UpcomingExpenses

module Sync =

    type SyncEntities =
        { Accounts: List<SyncAccount>
          Categories: List<SyncCategory>
          Transactions: List<SyncTransaction>
          UpcomingExpenses: List<SyncUpcomingExpense>
          Debts: List<SyncDebt>
          AutomaticTransactions: List<SyncAutomaticTransaction> }

File: src/Accountant/accountant2/src/lib/models/server/requests/sync.ts
import type { Category } from '$lib/models/entities/category';
import type { Account } from '$lib/models/entities/account';
import type { TransactionModel } from '$lib/models/entities/transaction';
import type { UpcomingExpense } from '$lib/models/entities/upcomingExpense';
import type { DebtModel } from '$lib/models/entities/debt';
import type { AutomaticTransaction } from '$lib/models/entities/automaticTransaction';

export class GetChanges {
	constructor(public lastSynced: string) {}
}

export class SyncEntities {
	constructor(
		public accounts: Account[],
		public categories: Category[],
		public transactions: TransactionModel[],
		public upcomingExpenses: UpcomingExpense[],
		public debts: DebtModel[],
		public automaticTransactions: AutomaticTransaction[]
	) {}
}
